src/query/api/v1/handler/prometheus/native/read_common.go

```diff
@@ -134,19 +134,25 @@ func sortedBlocksToSeriesList(blockList []blockWithMeta) ([]*ts.Series, error) {
 		return emptySeriesList, nil
 	}
 
-	firstBlock := blockList[0].block
+	var (
+		firstBlock = blockList[0].block
+		meta       = firstBlock.Meta()
+		bounds     = meta.Bounds
+		commonTags = meta.Tags.Tags
+	)
+
 	firstSeriesIter, err := firstBlock.SeriesIter()
 	if err != nil {
 		return nil, err
 	}
 
-	numSeries := firstSeriesIter.SeriesCount()
-	seriesMeta := firstSeriesIter.SeriesMeta()
-	bounds := firstSeriesIter.Meta().Bounds
-	commonTags := firstSeriesIter.Meta().Tags.Tags
+	var (
+		numSeries   = firstSeriesIter.SeriesCount()
+		seriesMeta  = firstSeriesIter.SeriesMeta()
+		seriesList  = make([]*ts.Series, numSeries)
+		seriesIters = make([]block.SeriesIter, len(blockList))
```

Comment:
Maybe we should opt for using `make(thing, 0, expectedLength)` and using `append(..)` wherever possible?

Reply:
Sounds good; I've tended to use indexed insertion when we know the slice length, but I guess for a fairly cold path it doesn't really matter and the safety is a bigger draw

Reply:
(especially considering that's literally the root cause of this PR haha)